lib/libzutil/os/linux/zutil_device_path_os.c

```diff
@@ -529,17 +529,19 @@ zfs_dev_is_whole_disk(const char *dev_name)
 {
 	struct dk_gpt *label;
```

Comment:
Unless I'm missing something, the current version of the function works fine and this is just some refactoring?  If so, I'd be inclined to leave it be, this function is small enough it's pretty clear what's going on.

Although, we should make a small change to ensure `label` is NULL initialized.